### ISSUE TITLE
Python 3 compatibility issue in from_url() parsing logic

### ISSUE THREAD

**Issue by megahall** [andymccurdy/redis-py#957]
Today I ran into the following issue in `redis==2.10.6`; it appears there is some occasional incompatibility between this bit of URL-handling code and Python 3 which is throwing an exception back to my client code. I did some searching but didn't see anyone else that ran into this yet.

```
> .../venv/lib/python3.6/site-packages/redis/connection.py(839)from_url()
--> 839         if '?' in url.path and not url.query:
    840             # chop the querystring including the ? off the end of the url

ipdb> url
ParseResultBytes(scheme=b'', netloc=b'', path=b'', params=b'', query=b'', fragment=b'')
ipdb> '?' in url.path
*** TypeError: a bytes-like object is required, not 'str'
ipdb>
```

**Comment by jeffbowen** [comment]
This same issue just popped up for me out of the blue. Not sure what could have changed.

**Comment by curaloucura** [comment]
I just migrated a project from python 3.5 to 3.6 and looks like the newer version is more strict

**Comment by gyermolenko** [comment]
hi @megahall 

Piece of code causing the issue (and the line `if '?' in url.path and not url.query:` in particular) was [removed](https://github.com/andymccurdy/redis-py/commit/5089cc275e6077a070017587001c22e490995d47#diff-b3be2d0fa4e0cf1f1c403f2de43ecdb4L841)

I believe the problem is not present since version 3.0

**Comment by andymccurdy** [comment]
Yes, this was fixed as part of 3.0. Sorry for not commenting about it in this issue when 3.0 was released.
